anndata/_core/anndata.py

```diff
@@ -422,16 +425,15 @@ def _init_as_actual(
 
             # init from DataFrame
             elif isinstance(X, pd.DataFrame):
+                # to verify index matching, we wait until obs and var are DataFrames
                 if obs is None:
                     obs = pd.DataFrame(index=X.index)
-                else:
-                    if not X.index.equals(obs.index):
-                        raise ValueError("Index of obs must match index of X.")
+                elif not isinstance(X.index, pd.RangeIndex):
```

Comment:
Why just `RangeIndex`? What about an integer valued index?

Reply:
Do we have a consistent behavior here over the whole codebase? If not we could of course introduce a function to use in these cases like `is_ignorable_index`.

Since that would touch many more places though, I feel like it’s a thing for another PR.